### ISSUE TITLE
False positive on LC0044 with InitPrimaryKeyFields equals false

### ISSUE THREAD

**Issue by Arthurvdv** [StefanMaron/BusinessCentral.LinterCop#838]
When using `TransferFields` between two tables, where only the PrimaryKeyFields of the tables are different, the rule shouldn't show an diagnostic when setting the boolean `InitPrimaryKeyFields` to false, see the example below.

```AL
ShippingUnit.TransferFields(ShipmentUnit, false); // Analyze only non-primary key fields
ShippingUnit.TransferFields(ShipmentUnit, true);  // Analyze all fields
```

## Example 

```AL
local procedure CreateShipmentUnit(SalesShipmentContainerNo: Code[20]; ShippingContainerNo: Code[20)
var
    ShippingUnit: Record "Shipping Unit PTE";
    ShipmentUnit: Record "Shipment Unit PTE";
begin
    ShippingUnit.SetRange("Shipping Container No.", ShippingContainerNo);
    if ShippingUnit.FindSet() then
        repeat
            Clear(ShipmentUnit);
            ShipmentUnit.Init();
            ShipmentUnit.Validate("Shipment Container No.", SalesShipmentContainerNo);
            ShippingUnit.TransferFields(ShipmentUnit, false); // LC0044: Records 'Shipping Unit PTE' and 'Shipment Unit PTE' have conflicting fields with the same ID
            ShipmentUnit.Insert(true);
        until ShippingUnit.Next() = 0;
end;
```

```AL
table 50100 "Shipping Unit PTE"
{
    Caption = 'Shipping Unit';

    fields
    {
        field(1; "Shipping Container No."; Code[20])
        {
            Caption = 'Shipping Container No.';
            TableRelation = "Shipping Container PTE"."No.";
        }
        field(2; "Line No."; Integer)
        {
            Caption = 'Line No.';
        }
...

table 50101 "Shipment Unit PTE"
{
    Caption = 'Shipment Unit', Comment = 'NLB="Verzendingseenheid"';
    DrillDownPageId = "Shipment Unit List VRA";
    LookupPageId = "Shipment Unit List VRA";

    fields
    {
        field(1; "Shipment Container No."; Code[20])
        {
            Caption = 'Shipment Container No.';
            TableRelation = "Sales Shipment Container PTE"."No.";
        }
        field(2; "Line No."; Integer)
        {
            Caption = 'Line No.';
        }
...
```
